tests/test_rabfe_topology.py

```diff
@@ -251,3 +254,51 @@ def test_dual_topology_rhfe():
     np.testing.assert_array_equal(combined_lambda_plane_idxs, np.zeros(C))
     np.testing.assert_array_equal(combined_lambda_offset_idxs[:A], np.zeros(A))
     np.testing.assert_array_equal(combined_lambda_offset_idxs[A:], np.ones(B))
+
+
+def test_dual_topology_charge_conversion():
+
+    # test charge parameter interpolation for a charged molecule
+
+    # ff = Forcefield.load_from_file("smirnoff_1_1_0_sc.py")
+    ff = Forcefield.load_from_file("smirnoff_1_1_0_ccc.py")
+
+    # (ytz): tbd change number of atoms in mol_a to catch offset errors
+
+    #                                      12  3 45  67 89 0 1  2    3
+    mol_a = Chem.AddHs(Chem.MolFromSmiles("CC(=O)OC1=CC=CC=C1C([O-])=O"))  # 20 atoms (inc. Hs)
+    mol_b = Chem.AddHs(Chem.MolFromSmiles("CC(-[O-])OC1=CC=CC=C1C(O)=O"))  # 22 atoms
+
+    AllChem.EmbedMolecule(mol_a)
+    AllChem.EmbedMolecule(mol_b)
+
+    num_a_atoms = mol_a.GetNumAtoms()
+
+    mol_top = topology.DualTopologyChargeConversion(mol_a, mol_b, ff)
+
+    qlj_params, nonbonded_potential = mol_top.parameterize_nonbonded(ff.q_handle.params, ff.lj_handle.params)
+
+    qlj_params_src = qlj_params[: len(qlj_params) // 2]
+    qlj_params_dst = qlj_params[len(qlj_params) // 2 :]
+
+    q_params_src = qlj_params_src[:, 0]
+    q_params_dst = qlj_params_dst[:, 0]
+
+    q_params_src_a = q_params_src[:num_a_atoms]
+    q_params_src_b = q_params_src[num_a_atoms:]
+
+    # np.testing.assert_array_equal(q_params_src_a, np.zeros_like(q_params_src_a))
+    np.testing.assert_array_equal(q_params_src_b, np.zeros_like(q_params_src_b))
+    # np.testing.assert_almost_equal(np.sum(q_params_src_b), -1 * np.sqrt(ONE_4PI_EPS0), decimal=5)
+    np.testing.assert_almost_equal(np.sum(q_params_src_a), -1 * np.sqrt(ONE_4PI_EPS0), decimal=5)
+
+    q_params_dst_a = q_params_dst[:num_a_atoms]
+    q_params_dst_b = q_params_dst[num_a_atoms:]
+
+    np.testing.assert_almost_equal(np.sum(q_params_dst_b), -1 * np.sqrt(ONE_4PI_EPS0), decimal=5)
+    np.testing.assert_array_equal(q_params_dst_a, np.zeros_like(q_params_dst_a))
+
+    # test that net charge along path of interpolation is -1
+    for lam in np.linspace(0, 1, 100):
+        qi = (1 - lam) * q_params_src + lam * q_params_dst
+        np.testing.assert_almost_equal(np.sum(qi), -1 * np.sqrt(ONE_4PI_EPS0), decimal=5)
```

Comment:
Question: Is there a way to assert this more directly about the behavior of NonbondedInterpolated?

(I can't immediately think of one, but would prefer if so.)

Reply:
I'm not sure either - maybe an identity transformation of two identical conformations of two identical molecules have the same energy? Seems a bit contrived though.

Reply:
Not sure, just it looked like this is closer to asserting that `sum(q_params_src) == sum(q_params_dst) ~= -sqrt(ONE_4PI_EPS0)` 100 times, rather than asserting that the interpolated potential is interpolating linearly between `q_params_src` and `q_params_dst`... I don't see a good way to improve upon the current test though!